src/lib/di/shared.hpp

```diff
@@ -0,0 +1,129 @@
+//
+// Copyright (c) 2012-2020 Kris Jusiak (kris at jusiak dot net)
+//
+// Distributed under the Boost Software License, Version 1.0.
+// (See accompanying file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+#pragma once
+
+#ifndef CANARY_SHARED_HPP
+	#define CANARY_SHARED_HPP
```

Comment:
you don't need this if you're already doing #pragma once

Reply:
Nice I didn't know that, I'll replace all for pragma